### PR TITLE
Branch b

### PR DESCRIPTION
**Pull request by dilsgeorge**
modified:

Dils:-
top_model/main.cpp
test/receiver/main.cpp
receiverCadmium.hpp

Namitha:-
test/sender/main.cpp
test/subnet/main.cpp
subnetCadmium.hpp

please review.

### CODE REVIEW

**Comment by mitalraithatha** [comment]
As per coding conventions all files name should be in small letters 
**Include->receiverCadmium.hpp**
**Include->subnetCadmium.hpp** 
change this files name to small letters.

**Comment by mitalraithatha** [comment]
All files should have less than 80 (<80)  line characters according to coding conventions.
Please apply that to all files.

**Comment by dilsgeorge** [comment]
done the changes, please review

File: include/receivercadmium.hpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
+/**
+* Cristina Ruiz Martin
+* ARSLab - Carleton University
+*
+*/
+
+#ifndef __BOOST_SIMULATION_PDEVS_RECEIVER_HPP__
```
Comment (mitalraithatha) [inline include/receivercadmium.hpp]: There is no problem in this header guard for compiling purpose but as a good practice according to naming rules if you change it to **__RECEIVER_CADMIUM_HPP__** then its better.

File: include/receivercadmium.hpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+   
+template<typename TIME>
+class Receiver{
+    using defs=receiver_defs; // putting definitions in context
+    public:
+    //Parameters to be overwriten when instantiating the atomic model
+    TIME   PREPARATION_TIME;
+    // default constructor
+    Receiver() noexcept{
+    	PREPARATION_TIME  = TIME("00:00:10");
```
Comment (mitalraithatha) [inline include/receivercadmium.hpp]: change the proper spacing before and after **=** sign.

File: include/receivercadmium.hpp [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                state.ack_num = static_cast < int > (x.value);
+                state.sending = true;
+            }
+                           
+    }
+
+    // confluence transition
+    void confluence_transition(TIME e,
+    	typename make_message_bags<input_ports>::type mbs) {
+        internal_transition();
```
Comment (mitalraithatha) [inline include/receivercadmium.hpp]: Indentation is missing here.
after curly braces there should be indentation in next line.

File: include/receivercadmium.hpp [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+    friend std::ostringstream& operator<<(std::ostringstream& os,
+    const typename Receiver<TIME>::state_type& i) {
+        os << "ackNum: " << i.ack_num;
+        return os;
+    }
+};
+  
+
+#endif // __BOOST_SIMULATION_PDEVS_RECEIVER_HPP__
```
Comment (mitalraithatha) [inline include/receivercadmium.hpp]: Same header guard naming rule should apply here.

File: include/receivercadmium.hpp [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        get_messages<typename defs::output>(bags).push_back(out);
+        return bags;
+    }
+
+    // time_advance function
+    TIME time_advance() const {
+        TIME next_internal;
+        if (state.sending) {
+            next_internal = PREPARATION_TIME;
+        }else {
```
Comment (mitalraithatha) [inline include/receivercadmium.hpp]: else will come in next line.
not on the same line.

File: include/subnetcadmium.hpp [hunk 6]
```diff
@@ -0,0 +1,119 @@
+/**
+* Cristina Ruiz Martin
+* ARSLab - Carleton University
+*
+*/
+
+#ifndef __BOOST_SIMULATION_PDEVS_SUBNET_HPP__
```
Comment (mitalraithatha) [inline include/subnetcadmium.hpp]: There is no problem in this header guard for compiling purpose but as a good practice according to naming rules if you change it to SUBNET_CADMIUM_HPP then its better.

File: include/subnetcadmium.hpp [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+            for (const auto &x : get_messages<typename defs::input>(mbs)){
+                state.packet = static_cast < int > (x.value);
+                state.transmiting = true;
+            }
+    }
+
+    // confluence transition
+    void confluence_transition(TIME e,
+        typename make_message_bags<input_ports>::type mbs) {
+        internal_transition();
```
Comment (mitalraithatha) [inline include/subnetcadmium.hpp]: Indentation is missing here, after curly braces in next line there should be indentation.

File: test/src/receiver/main.cpp [hunk 8]
```diff
@@ -16,56 +16,70 @@
...
-public:
-  ApplicationGen() = default;
-  ApplicationGen(const char* file_path) : iestream_input<Message_t,T>(file_path) {}
-};
+class ApplicationGen : public iestream_input<Message_t,T>{
+    public:
+    ApplicationGen() = default;
+    ApplicationGen(const char* file_path) : iestream_input<Message_t,
+        T>(file_path) {}
+    };
```
Comment (mitalraithatha) [inline test/src/receiver/main.cpp]: Remove 4 spaces from this line.
curly braces will end at the beginning of line as class Application is starting from beginning of line.

File: include/receivercadmium.hpp (hunk 2 above)
Comment (Namithajacob) [reply include/receivercadmium.hpp]: Is it mandatory?

File: test/src/sender/main.cpp [hunk 9]
```diff
@@ -27,48 +27,65 @@ using TIME = NDTime;
 
 
 /***** SETING INPUT PORTS FOR COUPLEDs *****/
-struct inp_controll : public cadmium::in_port<Message_t>{};
```
Comment (SREEJITH1991) [inline test/src/sender/main.cpp:30]: corrrect the structure name here

File: test/src/sender/main.cpp [hunk 10]
```diff
@@ -27,48 +27,65 @@ using TIME = NDTime;
...
-struct inp_ack : public cadmium::in_port<Message_t>{};
+struct input_control : public cadmium::in_port<Message_t>{};
+struct input_ack : public cadmium::in_port<Message_t>{};
 
 /***** SETING OUTPUT PORTS FOR COUPLEDs *****/
-struct outp_ack : public cadmium::out_port<Message_t>{};
```
Comment (SREEJITH1991) [inline test/src/sender/main.cpp:34]: give meaningful self explanatory names to  the structures

File: test/src/sender/main.cpp [hunk 11]
```diff
@@ -27,48 +27,65 @@ using TIME = NDTime;
...
-struct outp_data : public cadmium::out_port<Message_t>{};
-struct outp_pack : public cadmium::out_port<Message_t>{};
+struct output_ack : public cadmium::out_port<Message_t>{};
```
Comment (SREEJITH1991) [inline test/src/sender/main.cpp]: change Message_t to message_t, as per naming convention
Note: you may have have to change the name in message.hpp as well

File: test/src/sender/main.cpp [hunk 12]
```diff
@@ -78,63 +95,83 @@ using logger_top=cadmium::logger::multilogger<log_messages, global_time>;
...
+        oports_TOP,
+        eics_TOP,
+        eocs_TOP,
+        ics_TOP
+    );
 
 ///****************////
 
-    auto elapsed1 = std::chrono::duration_cast<std::chrono::duration<double, std::ratio<1>>>(hclock::now() - start).count();
+    auto elapsed1 = std::chrono::duration_cast<std::chrono::duration<double,
```
Comment (SREEJITH1991) [inline test/src/sender/main.cpp]: give meaningful name here

File: test/src/sender/main.cpp [hunk 13]
```diff
@@ -78,63 +95,83 @@ using logger_top=cadmium::logger::multilogger<log_messages, global_time>;
...
+    const char * i_input_data_control = input_data_control.c_str();
 
-std::shared_ptr<cadmium::dynamic::modeling::model> generator_con = cadmium::dynamic::translate::make_dynamic_atomic_model<ApplicationGen, TIME, const char* >("generator_con" , std::move(i_input_data_control));
+    std::shared_ptr<cadmium::dynamic::modeling::model> generator_con =
+        cadmium::dynamic::translate::make_dynamic_atomic_model<ApplicationGen,
+	    TIME, const char* >("generator_con" ,std::move(i_input_data_control));
 
-string input_data_ack = "test/data/sender/sender_input_test_ack_In.txt";
-const char * i_input_data_ack = input_data_ack.c_str();
+    string input_data_ack = "test/data/sender/sender_input_test_ack_In.txt";
```
Comment (SREEJITH1991) [inline test/src/sender/main.cpp]: all the input and output paths are hard coded here, we prefer soft coding . Using #define to represent the text would be a better way!!
Please check for other files as well

File: test/src/sender/main.cpp [hunk 14]
```diff
@@ -78,63 +95,83 @@ using logger_top=cadmium::logger::multilogger<log_messages, global_time>;
...
+    cadmium::dynamic::modeling::Ports iports_TOP = {};
+    cadmium::dynamic::modeling::Ports oports_TOP = {
+        typeid(output_data),typeid(output_pack),typeid(output_ack)
+    };
+    cadmium::dynamic::modeling::Models submodels_TOP = {
+        generator_con, generator_ack, sender1
+    };
+    cadmium::dynamic::modeling::EICs eics_TOP = {};
+    cadmium::dynamic::modeling::EOCs eocs_TOP = {
+        cadmium::dynamic::translate::make_EOC<Sender_defs::packetSentOut,
```
Comment (SREEJITH1991) [inline test/src/sender/main.cpp]: consider changing the structure names to sender_defs , packet_sent_output throughout

File: test/src/subnet/main.cpp [hunk 15]
```diff
@@ -74,55 +91,65 @@ using logger_top=cadmium::logger::multilogger<log_messages, global_time>;
 /********************************************/
 /****** APPLICATION GENERATOR *******************/
 /********************************************/
-string input_data = "test/data/subnet/subnet_input_test.txt";
-const char * i_input_data = input_data.c_str();
```
Comment (SREEJITH1991) [inline test/src/subnet/main.cpp:78]: a pointer variable should start with p_<variable_name>

**Review by SREEJITH1991 — CHANGES_REQUESTED** [review]
please review

File: test/src/sender/main.cpp (hunk 9 above)
Comment (dilsgeorge) [reply test/src/sender/main.cpp:30]: structure names were already changed in the first commit

File: test/src/sender/main.cpp (hunk 10 above)
Comment (dilsgeorge) [reply test/src/sender/main.cpp:34]: structure names were already changed in the first commit

File: test/src/sender/main.cpp (hunk 11 above)
Comment (dilsgeorge) [reply test/src/sender/main.cpp]: Message_t has been changed after review

File: test/src/sender/main.cpp (hunk 12 above)
Comment (dilsgeorge) [reply test/src/sender/main.cpp]: changed after review

File: test/src/sender/main.cpp (hunk 13 above)
Comment (dilsgeorge) [reply test/src/sender/main.cpp]: changed after review

File: test/src/sender/main.cpp (hunk 14 above)
Comment (dilsgeorge) [reply test/src/sender/main.cpp]: changed after review

File: test/src/subnet/main.cpp (hunk 15 above)
Comment (dilsgeorge) [reply test/src/subnet/main.cpp:78]: changed after review

**Comment by dilsgeorge** [comment]
kindly review the updated code .

**Review by mitalraithatha — APPROVED** [review]
Requested review updated.

File: test/src/sender/main.cpp (hunk 9 above)
Comment (SREEJITH1991) [reply test/src/sender/main.cpp:30]: verified

File: test/src/sender/main.cpp (hunk 10 above)
Comment (SREEJITH1991) [reply test/src/sender/main.cpp:34]: verified

Comment (SREEJITH1991) [reply test/src/sender/main.cpp:34]: verified

File: test/src/sender/main.cpp (hunk 12 above)
Comment (SREEJITH1991) [reply test/src/sender/main.cpp]: verified

File: test/src/sender/main.cpp (hunk 14 above)
Comment (SREEJITH1991) [reply test/src/sender/main.cpp]: verified

File: test/src/subnet/main.cpp (hunk 15 above)
Comment (SREEJITH1991) [reply test/src/subnet/main.cpp:78]: verified

File: test/src/sender/main.cpp (hunk 13 above)
Comment (SREEJITH1991) [reply test/src/sender/main.cpp]: verified

**Review by SREEJITH1991 — APPROVED** [review]

**Pull request merged**